src/Illuminate/Cache/DatabaseLock.php

```diff
@@ -77,13 +77,33 @@ public function acquire()
             $acquired = $updated >= 1;
         }
 
-        if (random_int(1, $this->lottery[1]) <= $this->lottery[0]) {
-            $this->connection->table($this->table)->where('expiration', '<=', time())->delete();
-        }
+        $this->clearExpiredLocks();
 
         return $acquired;
     }
 
+    /**
+     * Attempt to steal an existing lock.
+     *
+     * @return bool
+     */
+    public function steal()
+    {
+        $updated = $this->connection->table($this->table)
+            ->where('key', $this->name)
+            ->where('expiration', '<=', time())
+            ->update([
+                'owner' => $this->owner,
+                'expiration' => $this->expiresAt(),
+            ]);
+
+        $stolen = $updated >= 1;
+
+        $this->clearExpiredLocks();
+
+        return $stolen;
```

Comment:
`return $updated >= 1;` no?

Reply:
It would be better, I'd prefer to leave this as-is because this function (steal) has the same structure as the one just before it (acquire).